### PR TITLE
HPCC-15759 Avoid malloc in forked process before exec.

### PR DESCRIPTION
**Pull request by jakesmith**
Can cause problems if TBB malloc proxy being used.

Signed-off-by: Jake Smith jake.smith@lexisnexis.com

### CODE REVIEW

**Comment by jakesmith** [comment]
@richardkchapman - please review

File: system/jlib/jthread.cpp [hunk 1]
```diff
@@ -1954,6 +1954,7 @@ protected: friend class PipeWriterThread;
             execvp(argv[0],argv);
             _exit(START_FAILURE);    // must be _exit!!     
         }
+        free(argv);
```
Comment (richardkchapman) [inline system/jlib/jthread.cpp]: Probably need to free argv[*] too ?

Comment (jakesmith) [inline system/jlib/jthread.cpp]: I don't think so...?
argv is malloc to include space for line + array elements.
The args are memcpy'd in and the argv[<n>] members point to it.

Comment (richardkchapman) [inline system/jlib/jthread.cpp]: So it is. Horrible code!

File: system/jlib/jthread.cpp [hunk 2]
```diff
@@ -1899,6 +1899,8 @@ protected: friend class PipeWriterThread;
         if (haserror)
             if (::pipe(errpipe)==-1)
                 throw makeOsException(errno);
+        unsigned argc;
+        char **argv=splitargs(prog,argc);
```
Comment (richardkchapman) [inline system/jlib/jthread.cpp]: Worth putting a comment about why this is done before the fork, or some bright spark will think they can optimize it in the future.

File: system/jlib/jthread.cpp (hunk 1 above)
Comment (jakesmith) [inline system/jlib/jthread.cpp]: yes, weird.

File: system/jlib/jthread.cpp (hunk 2 above)
Comment (jakesmith) [inline system/jlib/jthread.cpp]: ok

**Pull request merged**